Make Beat role name version-independent for Community Beats (#3432)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/beat.asciidoc

ECK supports the deployment of any Community Beat. `type` and `version` specification elements have to be provided. In addition:

1. `image` element in the specification must point to the image to be deployed

> 2. the following roles must exist in Elasticsearch:ave the permissions required by the Bea

Reviewer: Editing error

> - if `elasticsearchRef` is provided, a role with `eck_beat_es_$type_role` name must exist, where `$type` is the Beat type. For example, when deploying `type: kafkabeat`, the role name would be `eck_beat_es_kafkabeat_role`. This role must have the permissions required by the Beat. See the link:https://www.elastic.co/guide/en/elasticsearch/reference/current/defining-roles.html[Elasticsearch documentation] for more details.

Reviewer: ```suggestion
- if `elasticsearchRef` is provided, a role with `eck_beat_es_$type_role` name must exist, where `$type` is the Beat type. For example, when deploying `kafkabeat`, the role name would be `eck_beat_es_kafkabeat_role`. This role must have the permissions required by the Beat. See the link:https://www.elastic.co/guide/en/elasticsearch/reference/current/defining-roles.html[Elasticsearch documentation] for more details.
```

Author: I wanted to to hint that this has to be the same as the type in the spec. Wouldn't this change hide that?

Reviewer: I think it's implied. Reading "...when deploying `type: kafkabeat`..." doesn't feel right to me but maybe it's fine. Perhaps @alaudazzi can chime in here.

> - if `kibanaRef` is provided, then, similarily to the above, a role with `eck_beat_kibana_$type_role` name must exist. This role must have the permissions required by the Beat for the dashboard setup purposes.

Reviewer: ```suggestion
- if `kibanaRef` is provided, then, as above, a role named `eck_beat_kibana_$type_role` must exist with the permissions required to setup Kibana dashboards.
```

> Alternatively to the 2. above, a user can be created in Elasticsearch and its credentials included in Beats `config` specification element  for Elasticsearch output, Kibana setup or both. If `elasticsearchRef` and `kibanaRef` are also used, ECK will still manage CA mounting and rollover, while using the provided user credentials.

Reviewer: ```suggestion
Alternatively,  create a user in Elasticsearch and include the credentials in the Beats `config` for Elasticsearch output, Kibana setup or both. If `elasticsearchRef` and `kibanaRef` are also defined, ECK will use the provided user credentials when setting up the connections.
```

ECK supports the deployment of any Community Beat. `type` and `version` specification elements have to be provided. In addition:

1. `image` element in the specification must point to the image to be deployed

> 2. the following roles must exist in Elasticsearch:ave the permissions required by the Bea

Reviewer: ```suggestion
2. the following roles must exist in Elasticsearch and have the permissions required by the Beat:
```

Author: Thanks! For some reason it didn't get to the first fix commit, fixed now.
